src/andershoest_ex/pa02/chutes_simulations.py

```diff
@@ -0,0 +1,257 @@
+        AH. note. Two first tests OK
+
+        """
+        self.board = board or Board()
+        random.seed(seed)
+        self.randomize_players = randomize_players
+        self.player_field = player_field or [Player, Player]
+        self.result = []
+
+    def single_game(self):
+        """
+        Runs a single game, returning a tuple consisting the number of moves
+        used to win the game and the type of the winner.
+
+        Using code from pa01
+
+        :param:
+
+        Making a list of player instances for the different player types.
+
+        """
+
+        player_list = [player_class(self.board)
+                       for player_class in self.player_field]
+        # Making a list of players of different classes in a single game
+        no_moves = 0
+        # All players make moves as long as the game is not finished
+        while True:
+            for player in player_list:
+                player.move()
+                no_moves += 1
+                if self.board.goal_reached(player.position):
+                    # Last line returning the number of moves of the class
+                    # instance that won the game,
+                    #  and the name of that type.
+                    return player.no_moves, type(player).__name__
+
+    def run_simulation(self, num_games=10):
+        """
+        run a given number of games and stores the results in the Simulation
+        object.
+        :param:
+
+        """
+        for _ in range(num_games):
+            self.result.append(self.single_game())
+
+    def get_results(self):
+        """
+        Return all the results generated by run_simulation as a tuple with
+        the number of games won by different types
+
+        """
+        return self.result
+
+    def winners_per_type(self):
+        """
+        Returns a dictionary mapping player types to the number of wins
+        with the key representing the player type and the value the
+        number of wins for that specific type.
+
+        :return:
+        """
+        winners = [winner[1] for winner in self.result]
+        # making a list of the type of winners
+        return Counter(winners)
+        # Using the Counter tool from the standard library to count the
+        # types in a dictionary
+
+
+    def durations_per_type(self):
+        """
+        Returning a dictionary mapping player types to lists of game durations
+        a dictionary of lists with the player type as key and the game
+        durations for that specific type as a list of values
+
+        """
+        pass
+
+
+if __name__ == "__main__":
+    sim = Simulation(player_field=[Player, LazyPlayer, ResilientPlayer])
+    sim.run_simulation(1)
+    results = sim.result
+    win_per_type = sim.winners_per_type()
+    win_list = [winner[1] for winner in sim.result]
+    board = Board()
+    player = Player(board)
+    #player.position = board.snake_dict[24]
+    player_temp_pos = 24
+    print(player.adjustment)
+
```

Comment:
pep8 extra blank line at the end of the file

Reply:
fixed